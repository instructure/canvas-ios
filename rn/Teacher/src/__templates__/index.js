//
// This file is part of Canvas.
// Copyright (C) 2018-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './users'
export * from './help-links'
export * from './dashboard'
export * from '../modules/assignee-picker/__template__/Assignee'
