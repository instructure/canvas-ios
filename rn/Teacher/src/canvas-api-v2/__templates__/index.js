// @flow
export * from './submission'
export * from './user'
export * from './assignment'
export * from './course'
export * from './group'
export * from './section'
export * from './groupSet'

// QUERIES
export * from '../queries/__templates__/SubmissionList'
export * from '../queries/__templates__/ContextCard'
