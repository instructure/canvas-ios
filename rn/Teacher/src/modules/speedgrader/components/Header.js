//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
  View,
  Image,
  TouchableHighlight,
} from 'react-native'
import { Text } from '../../../common/text'
import i18n from 'format-message'
import type {
  SubmissionDataProps,
} from '../../submissions/list/submission-prop-types'
import OldSubmissionStatusLabel from '../../submissions/list/OldSubmissionStatusLabel'
import Avatar from '../../../common/components/Avatar'
import { isAssignmentAnonymous } from '../../../common/anonymous-grading'
import { submissionTypeIsOnline } from '@common/submissionTypes'
import icon from '../../../images/inst-icons'
import { colors, createStyleSheet } from '../../../common/stylesheet'

export class Header extends Component<HeaderProps, State> {
  state: State = {
    showingPicker: false,
  }

  navigateToContextCard = () => {
    this.props.navigator.show(
      `/courses/${this.props.courseID}/users/${this.props.userID}`,
      { modal: true }
    )
  }

  navigateToPostPolicies = () => {
    this.props.navigator.show(`/courses/${this.props.courseID}/assignments/${this.props.assignmentID}/post_policy`, {
      modal: true,
    })
  }

  renderDoneButton () {
    return (
      <View style={styles.barButton}>
        <TouchableHighlight onPress={this.props.closeModal} underlayColor={colors.backgroundLightest} testID='header.navigation-done'>
          <Text style={{ color: colors.linkColor, fontSize: 18, fontWeight: '600' }}>
            {i18n('Done')}
          </Text>
        </TouchableHighlight>
      </View>
    )
  }

  renderEyeBall () {
    return (
      <View style={styles.barButton}>
        <TouchableHighlight onPress={this.navigateToPostPolicies} underlayColor={colors.backgroundLightest} testID='header.navigation-eye'>
          <View style={{ paddingLeft: 20 }}>
            <Image source={icon('eye', 'line')} style={styles.eyeIcon} />
          </View>
        </TouchableHighlight>
      </View>
    )
  }

  renderGroupProfile () {
    const sub = this.props.submissionProps
    let name = this.props.anonymous
      ? (sub.groupID ? i18n('Group') : i18n('Student'))
      : sub.name

    let avatarURL = this.props.anonymous
      ? ''
      : sub.avatarURL

    let action = this.navigateToContextCard
    let testID = `header.context.button.${this.props.userID}`
    if (sub.groupID && !this.props.anonymous) {
      action = this.showGroup
      testID = `header.groupList.button.${sub.groupID}`
    }
    const onlineSubmissionType = this.props.assignmentSubmissionTypes?.every(submissionTypeIsOnline)

    return (
      <View style={styles.profileContainer}>
        <TouchableHighlight
          style={styles.innerRowContainer}
          onPress={action}
          underlayColor={colors.backgroundLightest}
          testID={testID}
        >
          <View style={styles.innerRowContainer}>
            <View style={styles.avatar}>
              <Avatar
                key={sub.userID}
                avatarURL={avatarURL}
                userName={name}
              />
            </View>
            <View style={styles.nameContainer}>
              <Text style={styles.name} accessibilityTraits='header' numberOfLines={1}>{name}</Text>
              <OldSubmissionStatusLabel
                status={sub.status}
                onlineSubmissionType={onlineSubmissionType}
              />
            </View>
          </View>
        </TouchableHighlight>
        {this.props.newGradebookEnabled && this.renderEyeBall()}
        {this.renderDoneButton()}
      </View>
    )
  }

  render () {
    return (
      <View style={[this.props.style, styles.header]}>
        {this.renderGroupProfile()}
      </View>
    )
  }

  showGroup = () => {
    this.props.navigator.show(
      `/groups/${this.props.submissionProps.groupID}/users`,
      { modal: true },
      { courseID: this.props.courseID }
    )
  }
}

const styles = createStyleSheet(colors => ({
  header: {
    backgroundColor: colors.backgroundLightest,
    marginTop: 16,
  },
  profileContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  innerRowContainer: {
    backgroundColor: colors.backgroundLightest,
    flexDirection: 'row',
    alignItems: 'center',
    flex: 1,
  },
  navButtonImage: {
    resizeMode: 'contain',
    tintColor: colors.textInfo,
  },
  avatar: {
    width: 40,
    height: 40,
    marginLeft: 16,
  },
  nameContainer: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    marginLeft: 12,
    flex: 1,
  },
  name: {
    fontSize: 16,
    fontWeight: '600',
  },
  status: {
    fontSize: 14,
  },
  barButton: {
    backgroundColor: colors.backgroundLightest,
    marginRight: 12,
  },
  eyeIcon: {
    width: 20,
    height: 20,
    tintColor: colors.textDark,
  },
}))

export function mapStateToProps (state: AppState, ownProps: RouterProps) {
  const { courseID, assignmentID } = ownProps
  const anonymous = isAssignmentAnonymous(state, courseID, assignmentID)

  return {
    anonymous,
  }
}

let Connected = connect(mapStateToProps)(Header)
export default (Connected: any)

type RouterProps = {
  courseID: string,
  assignmentID: string,
  userID: string,
  submissionID: ?string,
  submissionProps: SubmissionDataProps,
  assignmentSubmissionTypes: SubmissionType[],
  closeModal: Function,
  style?: Object,
}

type State = {
  showingPicker: boolean,
}

type HeaderDataProps = {
  anonymous: boolean,
}

type HeaderProps = RouterProps & HeaderDataProps & { navigator: Navigator }
