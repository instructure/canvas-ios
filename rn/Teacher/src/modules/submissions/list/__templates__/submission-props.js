//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
