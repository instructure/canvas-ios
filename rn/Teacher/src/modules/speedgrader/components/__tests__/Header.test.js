//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* eslint-disable flowtype/require-valid-file-annotation */

import { shallow } from 'enzyme'
import _ from 'lodash'
import React from 'react'
import { Header, mapStateToProps } from '../Header'
import * as templates from '../../../../__templates__'

let noSubProps = {
  submissionID: null,
  assignmentID: '2',
  courseID: '3',
  userID: '4',
  assignmentSubmissionTypes: [ 'online_upload' ],
  submissionProps: {
    name: 'Allura',
    avatarURL: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
    status: 'none',
    userID: '4',
    grade: 'not_submitted',
    submissionID: null,
    submission: null,
  },
  closeModal: jest.fn(),
  excuseAssignment: jest.fn(),
  gradeSubmission: jest.fn(),
  selectSubmissionFromHistory: jest.fn(),
  selectedIndex: null,
  selectedAttachmentIndex: null,
  anonymous: false,
  navigator: templates.navigator(),
  newGradebookEnabled: false,
}

let subProps = {
  ...noSubProps,
  submissionID: '1',
  submissionProps: {
    name: 'Allura',
    avatarURL: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
    status: 'submitted',
    userID: '4',
    grade: '5',
    submissionID: '1',
    submission: templates.submissionHistory([
      { id: '1', grade: null, submitted_at: '2017-04-26T17:46:00Z' },
      { id: '2', grade: null, submitted_at: '2016-01-01T00:01:00Z' },
    ]),
  },
}

let groupProps = {
  ...subProps,
  submissionProps: {
    ...subProps.submissionProps,
    groupID: '1',
  },
}

describe('SpeedGraderHeader', () => {
  beforeEach(() => jest.clearAllMocks())

  it('renders with no submission', () => {
    let tree = shallow(<Header {...noSubProps} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders with a submission', () => {
    let tree = shallow(<Header {...subProps} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders with a group submission', () => {
    let tree = shallow(<Header {...groupProps} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders with a grade-only submission', () => {
    let props = _.cloneDeep(subProps)
    props.submissionProps.status = 'none'

    let tree = shallow(<Header {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders with an over-due grade-only submission', () => {
    let props = _.cloneDeep(subProps)
    props.submissionProps.status = 'missing'
    props.submissionProps.submission = null

    let tree = shallow(<Header {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders with an ungraded submission', () => {
    let props = _.cloneDeep(subProps)
    props.submissionProps.status = 'missing'
    props.assignmentSubmissionTypes = [ 'not_graded' ]

    let tree = shallow(<Header {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('anonymizes the avatar and name', () => {
    let tree = shallow(<Header {...subProps} anonymous />)
    expect(tree).toMatchSnapshot()
  })

  it('closes the modal', () => {
    let tree = shallow(<Header {...subProps} />)
    tree.find('[testID="header.navigation-done"]').simulate('Press')
    expect(subProps.closeModal).toHaveBeenCalled()
  })

  it('navigates to the context card when name is pressed', () => {
    let tree = shallow(<Header {...subProps} />)
    tree.find('[testID="header.context.button.4"]').simulate('Press')
    expect(subProps.navigator.show).toHaveBeenCalledWith(
      `/courses/3/users/4`,
      { modal: true },
    )
  })

  it('doesnt show the group name when anonymous', () => {
    let tree = shallow(<Header {...groupProps} anonymous />)
    expect(tree).toMatchSnapshot()
  })

  it('opens student list when group is tapped', () => {
    let tree = shallow(<Header {...groupProps} />)
    tree.find('[testID="header.groupList.button.1"]').simulate('Press')
    expect(groupProps.navigator.show).toHaveBeenCalledWith(
      `/groups/1/users`,
      { modal: true },
      { courseID: '3' },
    )
  })

  it('renders the eye icon when newGradebook is enabled', () => {
    let tree = shallow(<Header {...subProps} newGradebookEnabled />)
    let eye = tree.find('[testID="header.navigation-eye"]')
    expect(eye.length).toEqual(1)
  })

  it('navigates to post policy when tapping on the eye icon', () => {
    let navigator = { show: jest.fn() }
    let tree = shallow(
      <Header
        {...subProps}
        navigator={navigator}
        newGradebookEnabled
      />
    )
    let eye = tree.find('[testID="header.navigation-eye"]')
    eye.simulate('press')
    expect(navigator.show).toHaveBeenCalledWith(
      '/courses/3/assignments/2/post_policy',
      { modal: true }
    )
  })
})

describe('mapStateToProps', () => {
  it('returns the correct data when there is no submission', () => {
    let state = templates.appState({
      entities: {
        assignments: {
          '2': {
            data: templates.assignment({ anonymize_students: true }),
          },
        },
        submissions: {
          '1': {
            submission: {},
            pending: 0,
            error: null,
            selectedIndex: 3,
          },
        },
        courses: {},
      },
    })

    let dataProps = mapStateToProps(state, noSubProps)
    expect(dataProps).toMatchObject({
      anonymous: true,
    })
  })

  it('returns the correct data when there is a submission', () => {
    let state = templates.appState({
      entities: {
        assignments: {
          '2': {
            data: templates.assignment({ anonymize_students: true }),
          },
        },
        submissions: {
          '1': {
            submission: {},
            pending: 0,
            error: null,
            selectedIndex: 3,
          },
        },
        courses: {},
      },
    })

    let dataProps = mapStateToProps(state, subProps)
    expect(dataProps).toMatchObject({
      anonymous: true,
    })
  })

  it('returns the correct data when the assignment is for an anonymous quiz', () => {
    let state = templates.appState({
      entities: {
        assignments: {
          '2': {
            data: templates.assignment({ id: '2', quiz_id: '1' }),
          },
        },
        quizzes: {
          '1': {
            data: templates.quiz({ id: '1', anonymous_submissions: true }),
          },
        },
        submissions: {
          '1': {
            submission: {},
            pending: 0,
            error: null,
            selectedIndex: 3,
          },
        },
        courses: {
          '3': {},
        },
      },
    })

    let dataProps = mapStateToProps(state, subProps)
    expect(dataProps).toMatchObject({
      anonymous: true,
    })
  })

  it('returns the correct data when the assignment has anonymous grading turned on', () => {
    let state = templates.appState({
      entities: {
        assignments: {
          '2': {
            data: templates.assignment({ id: '2', anonymize_students: true }),
          },
        },
        submissions: {
          '1': {
            submission: {},
            pending: 0,
            error: null,
            selectedIndex: 3,
          },
        },
        courses: {
          '3': {},
        },
      },
    })

    let dataProps = mapStateToProps(state, subProps)
    expect(dataProps).toMatchObject({
      anonymous: true,
    })
  })
})
