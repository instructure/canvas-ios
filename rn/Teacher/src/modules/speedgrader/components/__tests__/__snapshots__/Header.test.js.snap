// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeedGraderHeader anonymizes the avatar and name 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FFFFFF",
        "marginTop": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="header.context.button.4"
      underlayColor="#FFFFFF"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 40,
              "marginLeft": 16,
              "width": 40,
            }
          }
        >
          <Avatar
            avatarURL=""
            key="4"
            userName="Student"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 12,
            }
          }
        >
          <Text
            accessibilityTraits="header"
            numberOfLines={1}
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Student
          </Text>
          <OldSubmissionStatusLabel
            onlineSubmissionType={true}
            status="submitted"
          />
        </View>
      </View>
    </TouchableHighlight>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[MockFunction]}
        testID="header.navigation-done"
        underlayColor="#FFFFFF"
      >
        <Text
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Done
        </Text>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader doesnt show the group name when anonymous 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FFFFFF",
        "marginTop": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="header.context.button.4"
      underlayColor="#FFFFFF"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 40,
              "marginLeft": 16,
              "width": 40,
            }
          }
        >
          <Avatar
            avatarURL=""
            key="4"
            userName="Group"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 12,
            }
          }
        >
          <Text
            accessibilityTraits="header"
            numberOfLines={1}
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Group
          </Text>
          <OldSubmissionStatusLabel
            onlineSubmissionType={true}
            status="submitted"
          />
        </View>
      </View>
    </TouchableHighlight>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[MockFunction]}
        testID="header.navigation-done"
        underlayColor="#FFFFFF"
      >
        <Text
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Done
        </Text>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with a grade-only submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FFFFFF",
        "marginTop": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="header.context.button.4"
      underlayColor="#FFFFFF"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 40,
              "marginLeft": 16,
              "width": 40,
            }
          }
        >
          <Avatar
            avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
            key="4"
            userName="Allura"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 12,
            }
          }
        >
          <Text
            accessibilityTraits="header"
            numberOfLines={1}
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Allura
          </Text>
          <OldSubmissionStatusLabel
            onlineSubmissionType={true}
            status="none"
          />
        </View>
      </View>
    </TouchableHighlight>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[MockFunction]}
        testID="header.navigation-done"
        underlayColor="#FFFFFF"
      >
        <Text
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Done
        </Text>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with a group submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FFFFFF",
        "marginTop": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="header.groupList.button.1"
      underlayColor="#FFFFFF"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 40,
              "marginLeft": 16,
              "width": 40,
            }
          }
        >
          <Avatar
            avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
            key="4"
            userName="Allura"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 12,
            }
          }
        >
          <Text
            accessibilityTraits="header"
            numberOfLines={1}
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Allura
          </Text>
          <OldSubmissionStatusLabel
            onlineSubmissionType={true}
            status="submitted"
          />
        </View>
      </View>
    </TouchableHighlight>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[MockFunction]}
        testID="header.navigation-done"
        underlayColor="#FFFFFF"
      >
        <Text
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Done
        </Text>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with a submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FFFFFF",
        "marginTop": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="header.context.button.4"
      underlayColor="#FFFFFF"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 40,
              "marginLeft": 16,
              "width": 40,
            }
          }
        >
          <Avatar
            avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
            key="4"
            userName="Allura"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 12,
            }
          }
        >
          <Text
            accessibilityTraits="header"
            numberOfLines={1}
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Allura
          </Text>
          <OldSubmissionStatusLabel
            onlineSubmissionType={true}
            status="submitted"
          />
        </View>
      </View>
    </TouchableHighlight>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[MockFunction]}
        testID="header.navigation-done"
        underlayColor="#FFFFFF"
      >
        <Text
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Done
        </Text>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with an over-due grade-only submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FFFFFF",
        "marginTop": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="header.context.button.4"
      underlayColor="#FFFFFF"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 40,
              "marginLeft": 16,
              "width": 40,
            }
          }
        >
          <Avatar
            avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
            key="4"
            userName="Allura"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 12,
            }
          }
        >
          <Text
            accessibilityTraits="header"
            numberOfLines={1}
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Allura
          </Text>
          <OldSubmissionStatusLabel
            onlineSubmissionType={true}
            status="missing"
          />
        </View>
      </View>
    </TouchableHighlight>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[MockFunction]}
        testID="header.navigation-done"
        underlayColor="#FFFFFF"
      >
        <Text
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Done
        </Text>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with an ungraded submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FFFFFF",
        "marginTop": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="header.context.button.4"
      underlayColor="#FFFFFF"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 40,
              "marginLeft": 16,
              "width": 40,
            }
          }
        >
          <Avatar
            avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
            key="4"
            userName="Allura"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 12,
            }
          }
        >
          <Text
            accessibilityTraits="header"
            numberOfLines={1}
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Allura
          </Text>
          <OldSubmissionStatusLabel
            onlineSubmissionType={false}
            status="missing"
          />
        </View>
      </View>
    </TouchableHighlight>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[MockFunction]}
        testID="header.navigation-done"
        underlayColor="#FFFFFF"
      >
        <Text
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Done
        </Text>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;

exports[`SpeedGraderHeader renders with no submission 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#FFFFFF",
        "marginTop": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      testID="header.context.button.4"
      underlayColor="#FFFFFF"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "height": 40,
              "marginLeft": 16,
              "width": 40,
            }
          }
        >
          <Avatar
            avatarURL="https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg"
            key="4"
            userName="Allura"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginLeft": 12,
            }
          }
        >
          <Text
            accessibilityTraits="header"
            numberOfLines={1}
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Allura
          </Text>
          <OldSubmissionStatusLabel
            onlineSubmissionType={true}
            status="none"
          />
        </View>
      </View>
    </TouchableHighlight>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "marginRight": 12,
        }
      }
    >
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[MockFunction]}
        testID="header.navigation-done"
        underlayColor="#FFFFFF"
      >
        <Text
          style={
            Object {
              "color": "#008EE2",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Done
        </Text>
      </TouchableHighlight>
    </View>
  </View>
</View>
`;
